system-contract-dapp-playground/src/api/hedera/tokenQuery-interactions/index.ts

```diff
@@ -0,0 +1,406 @@
+  } else if (serialNumber && serialNumber < 0) {
+    console.error('Invalid serial number');
+    return { err: 'Invalid serial number' };
+  }
+
+  // prepare events map
+  const eventMaps = {
+    TOKEN_INFO: 'TokenInfo',
+    FUNGIBLE_INFO: 'FungibleTokenInfo',
+    NON_FUNFIBLE_INFO: 'NonFungibleTokenInfo',
+  };
+
+  // invoking contract methods
+  try {
+    let transactionResult;
+    switch (API) {
+      case 'TOKEN_INFO':
+        // prepare transaction
+        transactionResult = await baseContract.getTokenInfoPublic(hederaTokenAddress);
+        break;
+      case 'FUNGIBLE_INFO':
+        // prepare transaction
+        transactionResult = await baseContract.getFungibleTokenInfoPublic(hederaTokenAddress);
+        break;
+      case 'NON_FUNFIBLE_INFO':
+        if (!serialNumber) {
+          console.error('Serial number is needed for querying NON_FUNGIBLE_INFO');
+          return { err: 'Serial number is needed for querying NON_FUNGIBLE_INFO' };
+        } else {
+          // prepare transaction
+          transactionResult = await baseContract.getNonFungibleTokenInfoPublic(
+            hederaTokenAddress,
+            serialNumber
+          );
+        }
+        break;
+    }
+
+    return await handleContractResponseWithDynamicEventNames(transactionResult, eventMaps, API);
+  } catch (err: any) {
+    console.error(err);
+    return { err, transactionHash: err.receipt && err.receipt.hash };
+  }
+};
+
+/**
+ * @dev queries token's certain information fields
+ *
+ * @dev integrates TokenQueryContract.getTokenDefaultFreezeStatusPublic()
+ *
+ * @dev integrates TokenQueryContract.getTokenDefaultKycStatusPublic()
+ *
+ * @dev integrates TokenQueryContract.getTokenCustomFeesPublic()
+ *
+ * @dev integrates TokenQueryContract.getTokenExpiryInfoPublic()
+ *
+ * @dev integrates TokenQueryContract.getTokenTypePublic()
+ *
+ * @dev integrates TokenQueryContract.getTokenKeyPublic()
+ *
+ * @param baseContract: ethers.Contract
+ *
+ * @param API: "FREEZE_STATUS" | "KYC_STATUS" | "CUSTOM_FEES" | "TOKEN_EXPIRY" | "TOKEN_TYPE" | "TOKEN_KEYS"
+ *
+ * @param hederaTokenAddress: string
+ *
+ * @param keyType?: IHederaTokenServiceKeyType
+ *
+ * @return Promise<TokenQuerySmartContractResult>
+ */
+export const queryTokenSpecificInfomation = async (
+  baseContract: Contract,
+  API:
+    | 'KYC_STATUS'
+    | 'TOKEN_TYPE'
+    | 'TOKEN_KEYS'
+    | 'CUSTOM_FEES'
+    | 'TOKEN_EXPIRY'
+    | 'FREEZE_STATUS',
+  hederaTokenAddress: string,
+  keyType?: IHederaTokenServiceKeyType
+): Promise<TokenQuerySmartContractResult> => {
+  // sanitize param
+  if (!isAddress(hederaTokenAddress)) {
+    console.error('Invalid token address');
+    return { err: 'Invalid token address' };
+  }
+
+  // prepare events map
+  const eventMaps = {
+    TOKEN_TYPE: 'TokenType',
+    TOKEN_KEYS: 'TokenKey',
+    CUSTOM_FEES: 'TokenCustomFees',
+    TOKEN_EXPIRY: 'TokenExpiryInfo',
+    KYC_STATUS: 'TokenDefaultKycStatus',
```

Comment:
For the kyc and freeze cases i don't believe you've getting the current status. Rather you're getting the default
```suggestion
    DEFAULT_KYC_STATUS: 'TokenDefaultKycStatus',
```